internal/server/iigointernal/base_speaker.go

```diff
@@ -0,0 +1,167 @@
+package iigointernal
+
+import (
+	"github.com/SOMAS2020/SOMAS2020/internal/common/gamestate"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/roles"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/rules"
+	"github.com/pkg/errors"
+	"math/rand"
+)
+
+type baseSpeaker struct {
+	id            int
+	budget        int
+	judgeSalary   int
+	ruleToVote    string
+	votingResult  bool
+	clientSpeaker roles.Speaker
+}
+
+func (s *baseSpeaker) withdrawJudgeSalary(gameState *gamestate.GameState) error {
+	var judgeSalary = int(rules.VariableMap["judgeSalary"].Values[0])
+	var withdrawError = WithdrawFromCommonPool(judgeSalary, gameState)
+	if withdrawError != nil {
+		featureSpeaker.judgeSalary = judgeSalary
+	}
+	return withdrawError
+}
+
+func (s *baseSpeaker) sendJudgeSalary() {
+	if s.clientSpeaker != nil {
+		amount, err := s.clientSpeaker.PayJudge()
+		if err == nil {
+			featureJudge.budget = amount
+			return
+		}
+	}
+	amount, _ := s.payJudge()
```

Comment:
What's being discarded?

Reply:
An error which is there to satisfy the interface function signature but we are 100% sure will always be nil at runtime if the code gets to that line

Reply:
just remove the error in the interface if so :)